core/src/main/java/org/apache/iceberg/rest/RESTUtil.java

```diff
@@ -271,4 +273,25 @@ public static String resolveEndpoint(String catalogUri, String endpointPath) {
   public static Map<String, String> configHeaders(Map<String, String> properties) {
     return RESTUtil.extractPrefixMap(properties, "header.");
   }
+
+  /**
+   * Generate a RFC 9562 UUIDv7 string.
+   *
+   * <p>Layout: - 48-bit Unix epoch milliseconds - 4-bit version (0b0111) - 12-bit random (rand_a) -
+   * 2-bit variant (RFC 4122, 0b10) - 62-bit random (rand_b)
+   */
+  public static String generateUuidV7() {
+    long epochMs = System.currentTimeMillis() & 0xFFFFFFFFFFFFL; // 48 bits
```

Comment:
I haven't reviewed this yet but just wanted to link how the JDK generates a UUIDv7 in JDK26: https://github.com/openjdk/jdk/commit/642ba4cfd18b7e17ff7f0ac3b2ce557bcaa93784

Reply:
I compared with JDK 26’s `UUID.ofEpochMillis` and our layout matches (48‑bit timestamp, version 7 nibble, RFC 4122 variant, 12+62 random bits). I’d like to keep a no‑arg `generateUuidV7()` to mirror `UUID.randomUUID()` and avoid exposing timestamps; since callers need a header value, it returns String. I’ll align by using `SecureRandom` and adding a 48‑bit range check. I won’t copy JDK code (licensing), but the behavior is equivalent. The test is also the same: verify `version() == 7` and `variant()==2`.